cpp/inc/bond/ext/grpc/unary_call.h

```diff
@@ -149,37 +159,58 @@ class unary_call final
     }
 
     /// @brief Get the server context for this call.
+    ///
+    /// @note May not be called on this instance if it has been moved-from.
     const grpc::ServerContext& context() const
     {
         BOOST_ASSERT(_impl);
         return _impl->_context;
     }
 
     /// @brief Get the server context for this call.
+    ///
+    /// @note May not be called on this instance if it has been moved-from.
     grpc::ServerContext& context()
     {
         BOOST_ASSERT(_impl);
         return _impl->_context;
     }
 
     /// @brief Get the request message for this call.
+    ///
+    /// @note May not be called on this instance if it has been moved-from.
     const TRequest& request() const
     {
         BOOST_ASSERT(_impl);
         return _impl->_request;
     }
 
     /// @brief Get the request message for this call.
+    ///
+    /// @note May not be called on this instance if it has been moved-from.
     TRequest& request()
     {
         BOOST_ASSERT(_impl);
         return _impl->_request;
     }
 
+    /// @brief Creates a std::shared_ptr<unary_call> from this instance.
+    ///
+    /// This instance may not be used after to_shared has been called; the
+    /// returned shared instance must be used instead.
+    ///
+    /// @note May not be called on this instance if it has been moved-from.
+    std::shared_ptr<unary_call> to_shared()
```

Comment:
Are we sure that we want to return a smart pointer instead of something like `shared_unary_call`? That will change the access semantics for the user.